src/js/jsx/shared/NumberInput.jsx

```diff
@@ -154,9 +156,9 @@ define(function (require, exports, module) {
             try {
                 var trimValue = _.trimRight(rawValue, this.props.suffix);
 
-                if (nls.commaDecimalSeparator) {
+                if (this.state.decimalSeparator === ",") {
                     // Replace decimal separator , with . and replace argument separator ; with ,
-                    trimValue.replace(new RegExp("\\,|\\;", "g"), function (match) {
+                    trimValue = trimValue.replace(new RegExp("\\,|\\;", "g"), function (match) {
```

Comment:
It seems like it could be nice to pull this out in to something like `nls.parseDecimal`